torch/functional.py

```diff
@@ -326,6 +328,71 @@ def parse_subscript(n: int) -> str:
 
     return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
 
+# Wrapper around _histogramdd and _histogramdd_bin_edges needed due to (Tensor, Tensor[]) return type.
+if TYPE_CHECKING:
+    # The JIT doesn't understand Union, so leave untyped for mypy
+    pass
+else:
+    def histogramdd(input: Tensor,
+                    bins: Union[List[Tensor], List[int]],
+                    range: Optional[List[float]] = None,
+                    weight: Optional[Tensor] = None,
+                    density: bool = False):
+        r"""
+        histogramdd(input, bins, *, range=None, weight=None, density=False, out=None) -> (Tensor, Tensor[])
+
+        Computes a histogram of the values in a tensor. Interprets the elements of an
+        input tensor with innermost dimension D as D-dimensional coordinates.
+
+        :attr:`bins` can be a sequence of D ints or a sequence of D 1D tensors.
+
+        If :attr:`bins` is a sequence of ints, it specifies the number of equal-width
+        bins in each dimension. By default, the lower and upper range of the bins is
+        determined by the minimum and maximum elements of the input tensor in the
+        corresponding dimension. The :attr:`range` argument can be provided to specify
+        ranges for the bins in each dimension.
+
+        If :attr:`bins` is a sequence of 1D tensors, it specifies the sequences of bin
+        edges, each including their rightmost edge. Each bin sequence should contain
+        an increasing sequence of at least 2 elements.
+
+        Args:
+            {input}
+            bins: int[] or Tensor[]. If int[], defines the number of equal-width bins
```

Comment:
bins (a sequence of integers or tensors): If a sequence of integers, ...

Reply:
This is divergent from NumPy, right? Because NumPy has bins=10 and this requires a list of the bins?

What's the reason for that divergence?

In what other ways is this divergent from NumPy? The test suggests there are a few more but I'm not sure I understand them

Reply:
NumPy additionally allows passing a single int specifying "The number of bins for all dimensions (nx=ny=…=bins)." Will add the same thing here.

The other divergence (as made apparent in the tests) is that NumPy doesn't flatten inputs with 3+ dimensions.

Reply:
Does it just error on inputs with 3 or more dimensions?

Reply:
Yes:

```
Traceback (most recent call last):
  File "/home/saketh/anaconda3/lib/python3.8/site-packages/numpy/lib/histograms.py", line 1015, in histogramdd
    N, D = sample.shape
ValueError: too many values to unpack (expected 2)
```